common/src/main/java/edu/snu/nemo/common/ir/Pipe.java

```diff
@@ -23,7 +23,7 @@
  * runtime-specific distributed data movement and storage mechanisms.
  * @param <O> output type.
  */
-public interface OutputCollector<O> extends Serializable {
+public interface Pipe<O> extends Serializable {
```

Comment:
IMHO, `OutputCollector` is a better name.
- `OutputCollector` explicitly tells you that it is used to collect outputs
- People may confuse this `Pipe` class with the `Pipe` execution property.

Storm/Flink also use `OutputCollector` as the name of the class for emitting outputs.
- Storm: https://github.com/apache/storm/blob/master/storm-client/src/jvm/org/apache/storm/task/OutputCollector.java
- Flink: https://github.com/apache/flink/blob/master/flink-runtime/src/main/java/org/apache/flink/runtime/operators/shipping/OutputCollector.java